src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -708,26 +733,28 @@ internal int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancella
                         {
                             bool emitPdbFile = Arguments.EmitPdb && emitOptions.DebugInformationFormat != Emit.DebugInformationFormat.Embedded;
 
-                            using (var peStreamProvider = new CompilerEmitStreamProvider(this, finalPeFilePath))
-                            using (var pdbStreamProviderOpt = emitPdbFile ? new CompilerEmitStreamProvider(this, finalPdbFilePath) : null)
+                            var peStreamProvider = new CompilerEmitStreamProvider(this, finalPeFilePath);
+                            var pdbStreamProviderOpt = emitPdbFile ? new CompilerEmitStreamProvider(this, finalPdbFilePath) : null;
+
+                            success = compilation.SerializeToPeStream(
+                                moduleBeingBuilt,
+                                peStreamProvider,
+                                pdbStreamProviderOpt,
+                                testSymWriterFactory: null,
+                                diagnostics: diagnosticBag,
+                                metadataOnly: emitOptions.EmitMetadataOnly,
+                                cancellationToken: cancellationToken);
+
+                            peStreamProvider.Close(diagnosticBag);
```

Comment:
Actually, let's use try-finally here. In case we have a bug in the compiler an an exception is thrown we should still dispose the streams. That way if the caller of our API is hardened against failures in Roslyn we would not leave streams open.